xds/server.go

```diff
@@ -315,11 +240,18 @@ func (s *GRPCServer) GracefulStop() {
 func routeAndProcess(ctx context.Context) error {
 	conn := transport.GetConnection(ctx)
 	cw, ok := conn.(interface {
-		VirtualHosts() []xdsresource.VirtualHostWithInterceptors
+		RoutingConfiguration() xdsresource.RoutingConfiguration
 	})
 	if !ok {
 		return errors.New("missing virtual hosts in incoming context")
 	}
+
+	rc := cw.RoutingConfiguration()
+	if rc.Err != nil { // Error out at routing l7 level, represents an nack before usable route configuration or resource not found for RDS or error combining LDS + RDS (Shouldn't happen).
+		logger.Errorf("RPC on connection with xDS Configuration error: %v", rc.Err)
+		return status.Error(codes.Unavailable, "error from xDS configuration for matched route configuration")
```

Comment:
This error string is very vague and does not give me (the application owner) much useful information about how to go about getting to the root cause of the issue.

Reply:
This is WAI. Disclosing information to the client is a security vulnerability about configuration on server side. Thus, the team discussed in the xDS Chat room to log server side, and return only a status code UNAVAILABLE to the client.